### ISSUE TITLE
Problem with removing old screenshots using --overwrite_screenshots 

### ISSUE THREAD

**Issue by rganeyev** [fastlane/fastlane#10032]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [X] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [X] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [X] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Deliver fails to remove old screenshot when I use --overwrite_screenshots options. 

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

```
INFO [2017-08-11 13:25:43.61]: Login successful

+-----------------------------+-------------------------+
|                deliver 2.53.1 Summary                 |
+-----------------------------+-------------------------+
| skip_binary_upload          | true                    |
| skip_metadata               | true                    |
| force                       | true                    |
| automatic_release           | true                    |
| overwrite_screenshots       | true                    |
| run_precheck_before_submit  | false                   |
| app_identifier              | com.laikaboss.motordrag |
| username                    | rganeyev@gmail.com      |
| screenshots_path            | ./screenshots           |
| metadata_path               | ./metadata              |
| edit_live                   | false                   |
| platform                    | ios                     |
| skip_screenshots            | false                   |
| skip_app_version_update     | false                   |
| submit_for_review           | false                   |
| precheck_default_rule_level | warn                    |
+-----------------------------+-------------------------+

INFO [2017-08-11 13:25:49.56]: Starting with the upload of screenshots...
INFO [2017-08-11 13:25:49.56]: Removing all previously uploaded screenshots...

Looking for related GitHub issues on fastlane/fastlane...
Search query: undefined method `each_with_index' for nil

URL: https://api.github.com/search/issues?q=undefined%20method%20%60each_with_index'%20for%20nil+repo:fastlane/fastlane
➡️  fix output of 'fastlane' for multiple platforms without default.
    https://github.com/fastlane/fastlane/pull/8302 [closed] 9 💬
    5 weeks ago

🔗  You can ⌘ + double-click on links to open them directly in your browser.
/usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.53.1/deliver/lib/deliver/upload_screenshots.rb:20:in `block in upload': [!] undefined method `each_with_index' for nil:NilClass (NoMethodError)
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.53.1/deliver/lib/deliver/upload_screenshots.rb:19:in `each'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.53.1/deliver/lib/deliver/upload_screenshots.rb:19:in `upload'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.53.1/deliver/lib/deliver/runner.rb:102:in `upload_metadata'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.53.1/deliver/lib/deliver/runner.rb:24:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.53.1/deliver/lib/deliver/commands_generator.rb:73:in `block (2 levels) in run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:178:in `call'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:153:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.5/lib/commander/runner.rb:476:in `run_active_command'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.53.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:64:in `run!'
	from /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.5/lib/commander/delegates.rb:15:in `run!'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.53.1/deliver/lib/deliver/commands_generator.rb:173:in `run'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.53.1/deliver/lib/deliver/commands_generator.rb:12:in `start'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.53.1/fastlane/lib/fastlane/cli_tools_distributor.rb:58:in `take_off'
	from /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.53.1/bin/fastlane:20:in `<top (required)>'
	from /usr/local/bin/fastlane:22:in `load'
	from /usr/local/bin/fastlane:22:in `<main>'
```

### Environment

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.5                                     |
| Ruby                        | 2.4.1                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.12.2                          |
| Installation Source         | /usr/local/bin/fastlane                     |
| Host                        | Mac OS X 10.12.5 (16F73)                    |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.4.1_1/lib          |
| OpenSSL Version             | OpenSSL 1.0.2l  25 May 2017                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.3                                       |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.53.1  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.2        |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.1        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.3.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.58.0       |
| faraday                   | 0.12.2       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| uber                      | 0.1.0        |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.3        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.12.0       |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.1        |
| bundler                   | 1.15.3       |
| faraday_middleware        | 0.12.2       |
| json                      | 2.1.0        |
| io-console                | 0.4.6        |
| openssl                   | 2.0.5        |
</details>

*generated on:* **2017-08-11**
</details>

**Comment by KrauseFx** [comment]
Thanks for reporting this. What's the best way to reproduce this problem? How do your local directories look like, and how does your iTC page look like?

**Comment by rganeyev** [comment]
I had a published app on the app store. I created a new version, initialized a directory using 
`fastlane deliver`.  My local directory looks like this:
![screen shot 2017-08-11 at 21 52 14](https://user-images.githubusercontent.com/694753/29227678-72ad5cbe-7edf-11e7-9e54-c80b25a312a6.png)

I updated some pictures (did not change the filenames, just photoshopped content).
When I tried to re-upload screenshots using 

`fastlane deliver --force --skip_metadata --skip_binary_upload --automatic_release --overwrite_screenshots --run_precheck_before_submit=false` 

I got an error from above. 

My itc page looks like this:
![screen shot 2017-08-11 at 22 03 04](https://user-images.githubusercontent.com/694753/29228042-e98c5adc-7ee0-11e7-9e9d-0a423c220c41.png)

Images for other languages are taken from the English. I have only iPhone 5.5' and iPad 12.9' screenshots ready.
